idea/idea-frontend-fir/src/org/jetbrains/kotlin/idea/frontend/api/fir/components/KtFirSmartcastProvider.kt

```diff
@@ -50,4 +54,22 @@ internal class KtFirSmartcastProvider(
             }?.let(::add)
         }
     }
+
+    override fun getStableNullability(expression: KtExpression): StableNullability {
```

Comment:
Please, wrap it with `withValidityAssertion { ... }` to check that current analysis session is still valid